src/brsTypes/BrsType.ts

```diff
@@ -90,6 +92,48 @@ export namespace ValueKind {
     }
 }
 
+/**
+ *  Converts a specified brightscript type in string into BrsType representation, with actual value
+ *  Note: only supports native types so far.  Objects such as array/AA aren't handled at the moment.
+ *  @param {string} type data type of field
+ *  @param {string} value optional value specified as string
+ *  @returns {BrsType} BrsType value representation of the type
+ */
+export function getBrsTypeFromString(type: string, value?: string): BrsType {
+    let returnValue: BrsType;
+
+    switch (type.toLowerCase()) {
+        case "boolean":
+            returnValue = value ? BrsBoolean.from(value === "true") : BrsBoolean.False;
+            break;
+        case "dynamic":
+            returnValue = BrsInvalid.Instance;
+            break;
+        case "integer":
+            returnValue = value ? new Int32(Number.parseInt(value)) : new Int32(0);
```

Comment:
I assume the parser will handle cases where value may not be a valid number? So it should be caught before this point I think..

Reply:
This is pulling straight from the XML value attribute, so there's no brightscript parsing before this point.